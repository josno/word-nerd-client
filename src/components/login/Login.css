.form {
	margin-left: auto;
	margin-right: auto;
	border: 2px solid black;
	height: fit-content;
	width: 50vw;
}

.form-tab,
.input-submit-button {
	list-style: none;
	display: inline-block;
	padding: 10px;
	border: 2px solid black;
	margin: 10px;
	font-size: 30px;
}

ul {
	padding: 0px;
}


a {
	text-decoration: none;
}


