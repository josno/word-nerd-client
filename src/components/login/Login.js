import React, { Component } from 'react';
import WordContext from '../../WordContext';
import { Link } from 'react-router-dom';
import TokenService from '../../services/token-service';
import AuthApiService from '../../services/auth-api-service';
import './Login.css';

class Login extends Component {
	static contextType = WordContext;
	constructor(props) {
		super(props);
		this.state = {
			username: '',
			password: ''
		};
		this.handleChange = this.handleChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	handleChange = e => {
		const { name, value } = e.target;
		this.setState({
			[name]: value
		});
	};

	handleSubmit = () => {
		const { username, password } = this.state;

		// TokenService.saveAuthToken(
		// 	TokenService.makeBasicAuthToken(username, password)
		// );

		AuthApiService.postLogin({
			user_name: username,
			password: password
		}) //passed in as one object
			.then(res => {
				// console.log(res.authToken);
				TokenService.saveAuthToken(res.authToken); //will only store the token when request is successful
				// this.props.onLoginSuccess();
			})
			.catch(res => {
				this.setState({ error: res.error });
			});
	};

	render() {
		return (
			<div className="login">
				<section>
					<div className="input-form">
						<input
							className="form-input"
							type="text"
							htmlFor="username"
							placeholder="username"
							value={this.state.username}
							name="username"
							onChange={e => this.handleChange(e)}
						/>
						<input
							className="form-input"
							type="password"
							htmlFor="password"
							placeholder="password"
							value={this.state.password}
							name="password"
							onChange={e => this.handleChange(e)}
						/>
					</div>

					<Link to="/game-home-page">
						<button
							className="input-submit-button"
							onClick={this.handleSubmit}
						>
							Login
						</button>
					</Link>
				</section>
			</div>
		);
	}
}

export default Login;
