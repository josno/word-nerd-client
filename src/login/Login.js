import React, { Component } from 'react';
import WordContext from '../WordContext';
import { Link } from 'react-router-dom';
import './Login.css';

class Login extends Component {
	static contextType = WordContext;

	handleSubmit = e => {
		const { username, password } = e.target;

		const token = window.btoa(username.value + ':' + password.value); //encoding

		window.localStorage.setItem('token', token);
	};

	render() {
		return (
			<div clasName="login">
				<section>
					<div className="input-form">
						<input
							className="form-input"
							type="text"
							htmlFor="username"
							placeholder="username"
							value={this.state.username}
							name="username"
							onChange={e => this.handleChange(e)}
						/>
						<input
							className="form-input"
							type="password"
							htmlFor="password"
							placeholder="password"
							value={this.state.password}
							name="password"
							onChange={e => this.handleChange(e)}
						/>
					</div>

					<Link to="/game-home-page">
						<button
							className="input-submit-button"
							onClick={this.handleSubmit}
						>
							Login
						</button>
					</Link>
				</section>
			</div>
		);
	}
}

export default Login;
