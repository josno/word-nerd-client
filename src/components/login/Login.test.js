import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import Login from './Login';
import renderer from 'react-test-renderer';

describe('<Login/>', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<Login />, div);
		ReactDOM.unmountComponentAtNode(div);
	});
	it('renders as expected', () => {
		const tree = renderer
			.create(
				<BrowserRouter>
					<Login />
				</BrowserRouter>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
