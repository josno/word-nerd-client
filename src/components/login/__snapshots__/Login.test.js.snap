// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> renders as expected 1`] = `
<div
  className="login"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="input-form"
    >
      <input
        aria-label="username"
        className="form-input"
        name="username"
        placeholder="username"
        required={true}
        type="text"
      />
      <input
        aria-label="password"
        className="form-input"
        name="password"
        placeholder="password"
        required={true}
        type="password"
      />
    </div>
    <button
      aria-label="login"
      className="input-submit-button"
      type="submit"
    >
      Login
    </button>
  </form>
  <div
    className="error-message"
  />
</div>
`;
